src/runtime/set.cpp

```diff
@@ -600,20 +642,121 @@ Box* setGt(BoxedSet* self, BoxedSet* rhs) {
     return setIssuperset(self, rhs);
 }
 
+Box* setContains(BoxedSet* self, Box* key) {
+    RELEASE_ASSERT(PyAnySet_Check(self), "");
+
+    if (PySet_Check(key)) {
+        try {
+            BoxAndHash k_hash(key);
+            return boxBool(self->s.find(k_hash) != self->s.end());
+        } catch (ExcInfo e) {
+            if (e.matches(TypeError))
+                PyErr_Clear();
```

Comment:
we don't need to call PyErr_Clear here, since we won't have set the exception as the CAPI exception (you have to do that manually).  maybe we should though... but anyway for now this could just be `if (!e.matches(TypeError)) throw e;`